src/util/StringUtils.h

```diff
@@ -508,26 +508,30 @@ J join(const vector<J>& to_join, const S& joiner) {
 
 // _____________________________________________________________________________
 inline string lstrip(const string& text, char c) {
-  size_t i = 0;
-  while (i < text.size() && text[i] == c) {
-    ++i;
+  auto it = text.begin();
```

Comment:
Can't get this to work as the `init` argument needs to match the joind types and e.g. the following

    template <typename J, typename S>
    J join(const vector<J>& to_join, const S& joiner) {
      auto op = [&joiner](const auto& a, const auto& b) {return a + joiner + b;};
      return std::accumulate(to_join.begin(), to_join.end(), J(joiner), op);
    }

breaks since there is no `std::basic_string(const char&)` constructor

Reply:
Ok, so first, you are setting the initial value to the joiner, this is not correct.
My version has the flaw that it would start with the joiner. So what we would need is probably:
template <typename J, typename S>
`J join(const vector<J>& to_join, const S& joiner) {
  if (to_join.empty) {return J();}
  auto op = [&joiner](const auto& a, const auto& b) {a + joiner + b;};
  return std::accumulate(to_join.begin() + 1, to_join.end(), to_join.first, op);
}`